datadriven/src/sgpp/datadriven/algorithm/GridFactory.cpp

```diff
@@ -26,32 +28,93 @@ using sgpp::base::algorithm_exception;
 namespace sgpp {
 namespace datadriven {
 
-sgpp::base::Grid *GridFactory::createGrid(sgpp::base::GeneralGridConfiguration& gridConfig,
-    std::vector<std::vector <size_t>> interactions) const {
-  Grid *grid;
-
-  if (gridConfig.type_ == GridType::ModLinear) {
-      grid = Grid::createModLinearGrid(gridConfig.dim_);
-    } else if (gridConfig.type_ == GridType::Linear) {
-      grid = Grid::createLinearGrid(gridConfig.dim_);
-    } else {
-      throw algorithm_exception(
-          "LearnerBase::InitializeGrid: An unsupported grid type was chosen!");
+sgpp::base::Grid *GridFactory::createGrid(const sgpp::base::GeneralGridConfiguration& gridConfig,
+    const std::vector<std::vector <size_t>> interactions) const {
+  Grid *tmpGrid;
+  if (gridConfig.type_ == GridType::Linear) {
+    tmpGrid = Grid::createLinearGrid(gridConfig.dim_);
+  } else if (gridConfig.type_ == GridType::LinearL0Boundary) {
+    tmpGrid = Grid::createLinearBoundaryGrid(
+        gridConfig.dim_, static_cast<base::GridPoint::level_type>(gridConfig.boundaryLevel_));
+  } else if (gridConfig.type_ == GridType::LinearBoundary) {
+    tmpGrid = Grid::createLinearBoundaryGrid(gridConfig.dim_);
+  } else if (gridConfig.type_ == GridType::ModLinear) {
+    tmpGrid = Grid::createModLinearGrid(gridConfig.dim_);
+  } else {
+    throw algorithm_exception(
+        "LearnerBase::InitializeGrid: An unsupported grid type was chosen!");
+  }
+
+  // Generate regular Grid with LEVELS Levels
+  if (interactions.size() == 0) {
+    tmpGrid->getGenerator().regular(gridConfig.level_);
+  } else {
+    std::cout << "Creating geometry aware sparse grid..." << std::endl;
+    tmpGrid->getGenerator().regularInter(gridConfig.level_, interactions, 0.0);
+    std::cout << "Interactions set!" << std::endl;
+  }
+  return tmpGrid;
+}
+
+std::vector<std::vector<size_t>> sgpp::datadriven::GridFactory::getInteractions(
+    std::string& stencil, std::vector<int64_t>& dim) const {
```

Comment:
stencil should be an enum.

Reply:
yes it can be, but it would lead in to changing every method again

Reply:
Do it please.